lib/rack/cache/meta_store.rb

```diff
@@ -34,11 +34,11 @@ def lookup(request, entity_store)
 
       # find a cached entry that matches the request.
       env = request.env
-      match = entries.detect{ |req,res| requests_match?((res['Vary'] || res['vary']), env, req) }
+      match = entries.detect{ |req,res| requests_match?((res['vary'] || res['vary']), env, req) }
       return nil if match.nil?
 
       _, res = match
-      if body = entity_store.open(res['X-Content-Digest'])
+      if body = entity_store.open(res['x-content-digest'])
```

Comment:
With this change, after deploying with the updated version of this gem, you can end up in a scenario where a long-term cached item still exists in the metastore with the capitalized header, but when it goes to get the key here using the lowercase header, it returns nil, and passes nil to the open method, resulting in an exception further down, and since on lookups on exceptions, it calls `pass` instead of `fetch`, it isn't going to store the response with the updated header and will continue to hit the error

Could check here for both cases, but I think another fine option is just letting people know they should flush their cache if they're upgrading from an earlier version to this one or later.

Reply:
Do you mind creating an issue for this?